packages/element/src/react-platform.native.js

```diff
@@ -8,27 +8,39 @@ import { omit } from 'lodash';
  * WordPress dependencies
  */
 import { applyFilters, doAction } from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { cloneElement } from './react';
 
-const render = ( element, id ) =>
-	AppRegistry.registerComponent( id, () => ( propsFromParent ) => {
-		const parentProps = omit( propsFromParent || {}, [ 'rootTag' ] );
+const render = ( element, id ) => {
+	class App extends Component {
+		constructor() {
+			super( ...arguments );
 
-		doAction( 'native.pre-render', parentProps );
+			const parentProps = omit( this.props || {}, [ 'rootTag' ] );
 
-		const filteredProps = applyFilters(
-			'native.block_editor_props',
-			parentProps
-		);
+			doAction( 'native.pre-render', parentProps );
 
-		doAction( 'native.render', filteredProps );
+			this.filteredProps = applyFilters(
+				'native.block_editor_props',
+				parentProps
+			);
+		}
 
-		return cloneElement( element, filteredProps );
-	} );
+		componentDidMount() {
+			doAction( 'native.render', this.filteredProps );
```

Comment:
I understand that the action `native.render` should happen after the component is rendered, so now this will be executed after the component is mounted, this means after the first render.